examples/file-feed/complete/src/browser-app/DataStore.js

```diff
@@ -0,0 +1,104 @@
+ // Need to include this to make webpack happy
+import { Buffer } from 'buffer/' // eslint-disable-line no-unused-vars
+import EventEmitter from 'events'
+import spawnNode from '../util/spawn-node'
+import OrbitDB from 'orbit-db'
+import { readFileContents } from './read-file'
+
+let instance
+
+class DataStore extends EventEmitter {
+  constructor (options) {
+    super()
+    spawnNode(options, (err, node) => {
+      if (err) {
+        console.log(err)
+      }
+      this.ipfs = node
+      this.emit('ready')
+    })
+  }
+
+  // Open an orbit-db database and hook up to the emitted events
+  openFeed (name) {
+    this.orbitdb = new OrbitDB(this.ipfs)
+    this.feed = this.orbitdb.feed(name, { cachePath: '/ipfd/ipfd2.db' })
+    this.feed.events.on('ready', () => this.emit('feed'))
+    this.feed.events.on('history', () => this.emit('update'))
+    this.feed.events.on('data', () => this.emit('update'))
+    if (this.timer) clearInterval(this.timer)
+    this.timer = setInterval(() => this._updatePeers(name), 1000)
+  }
+
+  getFile (hash) {
+    return new Promise((resolve, reject) => {
+      // Cat the contents of a multihash
+      this.ipfs.files.cat(hash)
+        .then((stream) => {
+          // Buffer all contents of the file as text
+          // and once buffered, create a blob for the picture
+          let buffer = []
+          let bytes = 0
+          stream.on('error', (e) => console.error(e))
+          stream.on('data', (d) => {
+            bytes += d.length
+            buffer.push(d)
+            this.emit('load', hash, bytes)
+          })
+          stream.on('end', () => {
+            this.emit('load', hash, bytes)
+            resolve(buffer)
+          })
```

Comment:
I would prefer not adding a library to abstract the underlaying functionality of the stream. After all, this is an example and people will have different approaches to how they use streams, some will use `bl` or `concat`, some won't., so keeping it clear what ipfs returns is clearer in this case imo.